tests/test_windows_mixins.py

```diff
@@ -0,0 +1,95 @@
+from hypothesis import given
+from hypothesis.strategies import composite, floats, integers
+
+from rasterio import windows
+from rasterio.coords import BoundingBox
+
+
+class FakeDataset(windows.WindowMethodsMixin):
+    """A test double to support testing of the mixin class"""
+
+    def __init__(self, **kwargs):
+        self.transform = kwargs.get('transform')
+        self.height = kwargs.get('height')
+        self.width = kwargs.get('width')
+
+    @property
+    def bounds(self):
+        a, b, c, d, e, f, _, _, _ = self.transform
+        return BoundingBox(c, f + e * self.height, c + a * self.width, f)
+
+    def __repr__(self):
+        return "<FakeDataset: transform={}, height={}, width={}>".format(self.transform, self.height, self.width)
+
+
+@composite
+def dataset_utm(draw):
```

Comment:
Pretty happy with this, I feel like I'm finally starting to wield hypothesis like it's meant to be wielded!